email_parser.py

```diff
@@ -78,10 +78,86 @@ def extract_articles(email):
         else:
             raise ValueError("No JSON array found in the response")
         
-        for article in articles:
-            if not all(key in article for key in ('title', 'description', 'url')):
-                raise ValueError(f"Invalid article structure: {article}")
-        
+        other_articles = [article for article in articles if article.get('is_other', True)]
+        if len(other_articles) > 0:
+            logger.info(f"Starting to crawl {len(other_articles)} other articles")
+            for article in other_articles:
+                summary = crawl_and_summarize(article.get('url'))
+                article['summary'] = summary
+
+        other_article_prompt_content = "__".join([f"{article.get('title')},{article.get('url')},{article.get('summary')}." for article in other_articles])
+        second_prompt = f"""
+                Analyze the following content and extract information about articles mentioned.
+                For each article:
+                1. Extract the original title, description (if available), and URL (look for [LINK: url] in the text)
+                2. Rewrite the title and description in a friendlier, lighter tone with a touch of personal feel
+                3. Keep the rewritten content concise and engaging
+                4. Restrict the description to be less than 160 characters, and be more to the point
```

Comment:
This is looking a little lengthy & hard to follow. Maybe we should extract the prompts to a service/utils for easier management

Reply:
Also I'm not sure why you name the variable `second_prompt`, `second_response`, etc. It's not meaningful at all